### PR TITLE
Support per-avatar gain adjustment on stereo input streams

### PR DESCRIPTION
**Pull request by kencooke**
This adds support for per-avatar gain adjustments (via the PAL sliders) for an avatar with a stereo input stream. The gain slider should behave the same, whether an avatar has a mono or stereo stream.

### Test Plan
1) Run the server from this PR as your local domain.
2) With Interface, join the domain and set your audio input device to stereo.
    This can be a stereo device, or a virtual device (using Virtual Audio Cable) to a media player.
    Stereo is enabled in the Audio menu, with the "Enable stereo input" checkbox.
3) Join again from another machine, and verify you can adjust the volume of the stereo stream. 
    The slider is in the People tab, in the list of avatars.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8700/app_12863_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8700/HighFidelity-Beta-PR12863-537e54e9d461f06c4601eeb97e68bb9120bcd935.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8700/HighFidelity-Beta-PR12863-537e54e9d461f06c4601eeb97e68bb9120bcd935.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Audio is clear and stereo. It can also be adjusted in the people app.

File: assignment-client/src/audio/AudioMixerSlave.cpp [hunk 1]
```diff
@@ -322,8 +322,14 @@ void AudioMixerSlave::addStream(AudioMixerClientData& listenerNodeData, const QU
...
-        for (int i = 0; i < AudioConstants::NETWORK_FRAME_SAMPLES_STEREO; ++i) {
-            _mixSamples[i] += float(streamPopOutput[i] * gain / AudioConstants::MAX_SAMPLE_VALUE);
+
+        // apply the avatar gain adjustment
+        auto& hrtf = listenerNodeData.hrtfForStream(sourceNodeID, streamToAdd.getStreamIdentifier());
+        gain *= hrtf.getGainAdjustment();
+
+        for (int i = 0; i < AudioConstants::NETWORK_FRAME_SAMPLES_PER_CHANNEL; i++) {
+            _mixSamples[2*i+0] += (float)streamPopOutput[2*i+0] * gain * (1/32768.0f);
+            _mixSamples[2*i+1] += (float)streamPopOutput[2*i+1] * gain * (1/32768.0f);
```
Comment (birarda) [inline assignment-client/src/audio/AudioMixerSlave.cpp]: `32768.0f` looks like a magic number we should const?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8753/app_12863_PR_2593d224c518f08a97e5ba41659759c1c2530268.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8753/HighFidelity-Beta-PR12863-2593d224c518f08a97e5ba41659759c1c2530268.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8753/HighFidelity-Beta-Interface-PR12863-2593d224c518f08a97e5ba41659759c1c2530268.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8753/HighFidelity-Beta-PR12863-2593d224c518f08a97e5ba41659759c1c2530268.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8753/HighFidelity-Beta-Interface-PR12863-2593d224c518f08a97e5ba41659759c1c2530268.exe

**Pull request merged**